bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/impl/QueryImpl.java

```diff
@@ -79,4 +81,13 @@ public T singleResult() {
           "Collection expected to have <1> entry but has <" + collection.size() + ">");
     }
   }
+
+  @Override
+  public Optional<T> findSingleResult() {
+    if (collection.size() == 1) {
+      return Optional.of(collection.iterator().next());
+    } else {
+      return Optional.empty();
+    }
+  }
```

Comment:
:x: The new method looks great. We can also refactor the code of the `singleResult()` method, to use the new `optionalSingleResult()`. It can look something like this:

```
  @Override
  public T singleResult() {
    if (optionalSingleResult().isPresent()) {
      return optionalSingleResult().get();
    } else {
      throw new BpmnModelException(
          "Collection expected to have <1> entry but has <" + collection.size() + ">");
    }
  }
```

Reply:
It's done.